src/tools/Avalonia.Generators/NameGenerator/AvaloniaNameSourceGenerator.cs

```diff
@@ -22,8 +22,16 @@ public void Execute(GeneratorExecutionContext context)
                 return;
             }
 
-            var partials = generator.GenerateNameReferences(context.AdditionalFiles);
-            foreach (var (fileName, content) in partials) context.AddSource(fileName, content);
+            var partials = generator.GenerateNameReferences(context.AdditionalFiles, context.CancellationToken);
+            foreach (var (fileName, content) in partials)
+            {
+                if(context.CancellationToken.IsCancellationRequested)
+                {
+                    break;
+                }
+
+                context.AddSource(fileName, content);
+            }
         }
         catch (Exception exception)
```

Comment:
OperactionCancelledException can be handled with no-op here. But it depends if file.GetText throws on cancellation or not

Reply:
Agreed. Fixed now